Evaluant.Calculator/Domain/EvaluationVisitor.cs

```diff
@@ -32,30 +33,18 @@ public override void Visit(LogicalExpression expression)
             throw new Exception("The method or operation is not implemented.");
         }
 
-        private static Type[] CommonTypes = new[] { typeof(Int64), typeof(Double), typeof(Boolean), typeof(String), typeof(Decimal) };
+        private static Type[] CommonTypes = { typeof(DateTime), typeof(Decimal), typeof(Double), typeof(Single), typeof(Int64), typeof(Int32), typeof(Int16), typeof(Boolean), typeof(String) };
```

Comment:
Good catch, I meant to point this out in my notes. I really wish I had a better understanding of the author's intent with how these are ordered, but it didn't strike me as "most specific to least specific", and it caused some test failures, i.e. comparing a string to an int resulted in `false`, but comparing a floating point number to a string threw an exception. (Or it may have been the other way around.) My order seemed to make more sense logically, and all the tests pass. So I _think_ it is a safe change and an improvement.